### ISSUE TITLE
remove CONFIG_OMIT_FRAME_POINTER from ARC boards

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1279]
**_Reported by Anas Nashif:_**

we need to revert the patch with Id:

Change-Id: I0901f55973c1ea37491b07bf625d0d1918803f3e

once the toolchain is fixed...

(Imported from Jira ZEP-1403)

**Comment by nashif** [comment]
**_by Sharron LIU:_**

Verify the said patch workaround the ARC GCC issue described in ZEP-1243.

```
00000000 <hmc5883l_sample_fetch>:
   0:   c0f1                    push_s  blink
   2:   1cfc b348               st.aw   r13,[sp,-4]
   6:   80a2                    ld_s    r13,[r0,0x8]
   8:   c1a8                    sub_s   sp,sp,0x20   // <- Stack increased to include local variables
   a:   40a5                    ld_s    r0,[r13,0]
   c:   c188                    add_s   r1,sp,0x20
   e:   734c                    mov_s   r2,3
  10:   19e1 808a               stb.aw  r2,[r1,-31]
  14:   c142                    st_s    r1,[sp,0x8]
  16:   712c                    mov_s   r1,1
  18:   420c                    ld_s    r2,[r0,0x4]
  1a:   c143                    st_s    r1,[sp,0xc]
  1c:   702c                    mov_s   r1,0
  1e:   c164                    stb_s   r1,[sp,0x10]
  20:   2440 3081               add     r1,sp,0x2
  24:   c145                    st_s    r1,[sp,0x14]
  26:   762c                    mov_s   r1,6
  28:   c146                    st_s    r1,[sp,0x18]
  2a:   d907                    mov_s   r1,0x7
  2c:   c167                    stb_s   r1,[sp,0x1c]
  2e:   8281                    ld_s    r12,[r2,0x4]
  30:   c182                    add_s   r1,sp,0x8
  32:   724c                    mov_s   r2,2
  34:   7c60                    jl_s.d  [r12]
  36:   db1e                    mov_s   r3,0x1e
  38:   0833 0012               brlt.nt r0,0,6a <hmc5883l_sample_fetch+0x6a>
  3c:   1402 3100               ldh     r0,[sp,2]
```

**Comment by nashif** [comment]
**_by Anas Nashif:_**

should be resolved with 0.9 SDK

**Comment by nashif** [comment]
**_by Juro Bystricky:_**

The ARC compiler has a fix for this now.

https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/cbd8e54244cd02bdcf4f1057be3ce96631f35ac3

The code with frame pointer looks correct:

{code:java}
40035440 <hmc5883l_sample_fetch>:
40035440:	c0f1                	push_s	blink
40035442:	1cfc b348           	st.aw	r13,[sp,-4]
40035446:	1cfc b6c8           	st.aw	fp,[sp,-4]
4003544a:	439b                	mov_s	fp,sp           // Frame pointer set
4003544c:	c1a8                	sub_s	sp,sp,0x20      // Stack adjusted
4003544e:	80a2                	ld_s	r13,[r0,0x8]
40035450:	4163                	mov_s	r1,fp
40035452:	40a5                	ld_s	r0,[r13,0]
40035454:	734c                	mov_s	r2,3
40035456:	19e1 808a           	stb.aw	r2,[r1,-31]
4003545a:	1be8 b040           	st	r1,[fp,-24]     // <- Frame Pointer used to access local variable
4003545e:	712c                	mov_s	r1,1
40035460:	420c                	ld_s	r2,[r0,0x4]
40035462:	1bec b040           	st	r1,[fp,-20]
40035466:	702c                	mov_s	r1,0
40035468:	1bf0 b042           	stb	r1,[fp,-16]
4003546c:	2342 3781           	sub	r1,fp,0x1e
40035470:	1bf4 b040           	st	r1,[fp,-12]
40035474:	762c                	mov_s	r1,6
40035476:	1bf8 b040           	st	r1,[fp,-8]
4003547a:	d907                	mov_s	r1,0x7
4003547c:	1bfc b042           	stb	r1,[fp,-4]
40035480:	8281                	ld_s	r12,[r2,0x4]
40035482:	db1e                	mov_s	r3,0x1e
40035484:	724c                	mov_s	r2,2
40035486:	7c60                	jl_s.d	[r12]
40035488:	2342 3601           	sub	r1,fp,0x18
4003548c:	780b                	tst_s	r0,r0
...
```

The fix will be included  in SDK 0.9

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Correcting the priority field

**Comment by nashif** [comment]
**_by Sharron LIU:_**

Verified the arc compiler issue fixed in Zephyr SDK v0.9, and the patch "CONFIG_OMIT_FRAME_POINTER=y" reverted from all ARC default configures.

Closing this. Thanks.
